src/plugins/memdump/userhook.cpp

```diff
@@ -719,6 +729,128 @@ void memdump::load_wanted_targets(const memdump_config* c)
     }
 }
 
+static event_response_t copy_on_write_ret_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    auto data = get_trap_params<memdump, copy_on_write_result_t<memdump>>(info);
+    memdump* plugin = data->plugin();
+    if (!data || !plugin)
+    {
+        PRINT_DEBUG("[MEMDUMP-USER] copy_on_write_ret_cb invalid trap params!\n");
+        drakvuf_remove_trap(drakvuf, info->trap, nullptr);
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    if (!data->verify_result_call_params(info, drakvuf_get_current_thread(drakvuf, info)))
+        return VMI_EVENT_RESPONSE_NONE;
+
+    plugin->destroy_trap(drakvuf, info->trap);
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    page_info_t pinfo;
+    addr_t pa;
+
+    if (vmi_pagetable_lookup_extended(vmi, info->regs->cr3, data->vaddr, &pinfo) != VMI_SUCCESS)
```

Comment:
`vmi_pagetable_lookup` should be enough here, you are not using anything else then the pa from `page_info_t`